File: C++/maxInWindows.cpp
#include<iostream>
#include<algorithm>
#include<vector>

class Solution {
public:
    std::vector<int> maxInWindows(std::vector<int>& num, int size) {
        if(size>num.size() || size == 0) return {};
        std::vector<int>result;
      
        for(int i=0;i<=num.size()-size;i++){
            std::vector<int>windows;
            for(int j =0;j<size;j++){
                windows.push_back(num[i+j]);
            }
            result.push_back(*std::max_element(windows.begin(),windows.end()));
        }
        return result;
    }
};

int main(){
    std::vector<int>nums={2,3,4,2,6,2,5,1};
    int size=3;
    Solution solution;
    std::vector<int>result=solution.maxInWindows(nums,size);
    std::cout<<"Max elements in windows :";
    for(int num:result){
        std::cout<<num<<" ";
    }
    std::cout<<std::endl;
    return 0;
}

File: python/maxInWindows.py
class Solution:
    def maxInWindows(self, num: list[int], size: int) -> list[int]:
        if size > len(num) or size == 0:
            return []

        result = []
        for i in range(len(num) - size + 1):
            windows = []
            for j in range(size):
                windows.append(num[i + j])
            result.append(max(windows))

        return result
    
num=[2,3,4,2,6,2,5,1]
size=3
s=Solution()
print(s.maxInWindows(num,size))